core/common/src/main/java/alluxio/underfs/UnderFileSystem.java

```diff
@@ -98,15 +102,17 @@ void clear() {
     private static class Key {
       private final String mScheme;
       private final String mAuthority;
+      private final Map<String, String> mProperties;
 
-      Key(AlluxioURI uri) {
+      Key(AlluxioURI uri, Map<String, String> properties) {
```

Comment:
as talked offline, in most of the cases, getting a ufs  is now a call keyed by mountId